pkg/controller/backup_controller.go

```diff
@@ -385,19 +385,21 @@ func (controller *backupController) runBackup(backup *api.Backup, bucket string)
 	} else {
 		backupSizeBytes = backupFileStat.Size()
 	}
-
-	if err := controller.backupService.UploadBackup(bucket, backup.Name, backupJSONToUpload, backupFileToUpload, logFile); err != nil {
-		errs = append(errs, err)
-	}
-
 	backupScheduleName := backup.GetLabels()["ark-schedule"]
-	controller.metrics.SetBackupTarballSizeBytesGauge(backupScheduleName, backupSizeBytes)
 
+	// Mark completion timestamp before uploading. Otherwise, the JSON file in object storage
```

Comment:
Can we organize lines 388-401 so that we (1) set the completion timestamp on the backup, (2) upload it, (3) record metrics (including creation of any vars needed to do so)? It's all a bit mixed together right now.